lib/IDE/CompletionInstance.cpp

```diff
@@ -618,6 +634,9 @@ bool swift::ide::CompletionInstance::performOperation(
   // the cached completion instance.
   std::lock_guard<std::mutex> lock(mtx);
 
+  // Clear the cached CI if it's invalidated.
+  clearCachedCompilerInstanceIfNeeded();
```

Comment:
Why was it necessary to introduce this extra function and not just add an additional check inside `performCachedOperationIfPossible()`?